pandas/tests/frame/test_reshape.py

```diff
@@ -1128,3 +1128,34 @@ def test_stack_timezone_aware_values():
         ),
     )
     tm.assert_series_equal(result, expected)
+
+
+def test_unstacking_multi_index_df():
+    # BUG: gh-30740
+    df = DataFrame(
+        {
+            "name": ["Alice", "Bob"],
+            "score": [9.5, 8],
+            "employed": [False, True],
+            "kids": [0, 0],
+            "gender": ["female", "male"],
+        }
+    )
+    df = df.set_index(["name", "employed", "kids", "gender"])
+    df = df.unstack(["gender"], fill_value=0)
+    expected = df.unstack("employed", fill_value=0).unstack("kids", fill_value=0)
+    result = df.unstack(["employed", "kids"], fill_value=0)
+    expected = DataFrame(
+        [[9.5, 0.0, 0.0, 0.0], [0.0, 0.0, 0.0, 8.0]],
+        Index(["Alice", "Bob"], name="name"),
+        columns= MultiIndex.from_tuples(
```

Comment:
you need to run black pandas before a push

```suggestion
        columns=MultiIndex.from_tuples(
```

Reply:
Will take care from next time